extensions/positron-r/src/provider.ts

```diff
@@ -25,123 +25,101 @@ export const R_DOCUMENT_SELECTORS = [
 	{ language: 'r', pattern: '**/*.{rprofile,Rprofile}' },
 ];
 
-/**
- * Enum represents the source from which an R binary was discovered.
- */
-enum BinarySource {
-	/* eslint-disable-next-line @typescript-eslint/naming-convention */
-	HQ = 'HQ',
-	adHoc = 'ad hoc location',
-	registry = 'Windows registry',
-	/* eslint-disable-next-line @typescript-eslint/naming-convention */
-	PATH = 'PATH',
-	user = 'user-specified directory'
+export interface RBinary {
+	path: string;
+	reasons: ReasonDiscovered[]
```

Comment:
Keeping an array of reasons for discovery now. For #5223, it should be possible for certain R installations to appear in more than 1 section of the picker, especially in a future with increased R support for environments, in the renv or conda sense.